kernel/src/platform/mod.rs

```diff
@@ -60,6 +60,43 @@ pub trait Platform {
     ) -> Result<(), returncode::ReturnCode> {
         Ok(())
     }
+
+    /// This function is called when an app faults
+    ///
+    /// This is an optional function that can be implemented by `Chip`s that
+    /// allows the chip to handle the app fault and not terminate or restart
+    /// the app.
+    ///
+    /// If `Ok(())` is returned by this function then the kernel will not
+    /// terminate or restart the app, but instead allow it to continue
+    /// running. NOTE in this case the chip must have fixed the underlying
+    /// reason for fault otherwise it will re-occur.
+    ///
+    /// This can not be used to get around Tock's protections. If for example
+    /// this function just ignored the error and allowed the app to continue
+    /// the fault would continue to occur,
+    ///
+    /// If `Err(())` is returned then the kernel will set the app as faulted
+    /// and follow the `FaultResponse` protocol.
+    ///
+    /// It is unlikey a `Chip` will need to implement this. This should be used
+    /// for only a handul of use cases. Possible use cases include:
+    ///    - Allowing the kernel to emulate un-implemented instructions
+    ///      This could be used to allow apps to run on hardware that doesn't
+    ///      implement some instructions, for example atomics.
+    ///    - Allow the kernel to handle hardware faults, triggered by the app
+    ///      This can allow an app to continue running if it triggers certain
+    ///      types of faults. For example if an app triggers a memory parity
+    ///      error the kernel can handle the error and allow the app to
+    ///      continue (or not).
+    ///    - Allow an app to execute from external QSPI
+    ///      This could be used to allow an app to execute from external QSPI
+    ///      where access faults can be handled by the `Chip` to ensure the
+    ///      QPSI is mapped correctly.
+    #[allow(unused_variables)]
+    fn process_app_fault(&self, process: &dyn process::ProcessType) -> Result<(), ()> {
```

Comment:
This name is a little confusing. Is process a noun or a verb here? Maybe `process_fault_hook()`?

Reply:
It was a verb, I have changed it to `process_fault_hook()`